docker/pgosm_flex.py

```diff
@@ -0,0 +1,318 @@
+"""Python script to run PgOSM Flex.
+
+Designed to be ran in Docker image:
+    https://hub.docker.com/r/rustprooflabs/pgosm-flex
+"""
+import click
+import subprocess
+import logging
+import sys
+import os
+import time
+import datetime
+import shutil
+import osm2pgsql_recommendation as rec
+
+
+BASE_PATH_DEFAULT = '/app'
+"""Default path for pgosm-flex project for Docker.
+"""
+
+def get_today():
+    today = datetime.datetime.today().strftime('%Y-%m-%d')
+    return today
+
+@click.command()
+@click.option('--layerset', required=True,
+              prompt='PgOSM Flex Layer Set',
+              help='Layer set from PgOSM Flex to load. e.g. run-all')
+@click.option('--ram', required=True,
+              prompt='Server RAM (GB)',
+              help='Amount of RAM in GB available on the server running this process.')
+@click.option('--region', required=True,
+              prompt="Region name",
+              help='Region name matching the filename for data sourced from Geofabrik. e.g. north-america/us')
+@click.option('--subregion', required=False,
+              default=None,
+              help='Sub-region name matching the filename for data sourced from Geofabrik. e.g. district-of-columbia')
+@click.option('--pgosm-date', required=False,
+              default=get_today(),
+              envvar="PGOSM_DATE",
+              help="Date of the data in YYYY-MM-DD format.")
+@click.option('--basepath',
+              required=False,
+              default=BASE_PATH_DEFAULT,
+              help='Used when testing locally and not within Docker')
+def run_pgosm_flex(layerset, ram, region, subregion, pgosm_date,
+                   basepath):
+    """Main logic to run PgOSM Flex within Docker.
+    """
+
+    paths = get_paths(base_path=basepath)
+    region_filename = get_region_filename(region, subregion)
+    """Matches Geofabrik naming conventions"""
+
+    log_file = get_log_path(region, subregion, paths)
+    print(f'Logging output to {log_file}')
+
+    # Docker image has Python 3.7, this doesn't work for some reason.
```

Comment:
the encoding parameter was added in Python 3.9, I think in this dockerized environment is utf-8 anyways